kart/help.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import os
+import subprocess
+import sys
+import platform
+import shlex
+from subprocess import PIPE
+from pathlib import Path
+
+from docutils.core import publish_string
+from docutils.writers import manpage
+from kart.docs import generate_man_page
+
+L = logging.getLogger("kart.help")
+
+
+class ExecutableNotFoundError(Exception):
+    def __init__(self, executable_name):
+        super(ExecutableNotFoundError, self).__init__(
+            'Could not find executable named "%s"' % executable_name
+        )
+
+
+def get_renderer():
+    """
+    Return the appropriate HelpRenderer implementation for the
+    current platform.
+    """
+    if platform.system() == "Linux" or platform.system() == "Darwin":
+        return PosixHelpRenderer()
+
+
+class PagingHelpRenderer(object):
+    """
+    Interface for a help renderer.
+
+    The renderer is responsible for displaying the help content on
+    a particular platform.
+    """
+
+    def __init__(self, output_stream=sys.stdout):
+        self.output_stream = output_stream
+
+    PAGER = None
+
+    def get_pager_cmdline(self):
+        pager = self.PAGER
+        if "MANPAGER" in os.environ:
+            pager = os.environ["MANPAGER"]
+        elif "PAGER" in os.environ:
+            pager = os.environ["PAGER"]
+        return shlex.split(pager)
+
+    def render(self, ctx, contents):
+        """
+        Each implementation of HelpRenderer must implement this
+        render method.
+        """
+        # we add a dummy header now and remove it after conversion
+        header = "KART\n====\n"
+        converted_content = self._convert_doc_content(ctx, header + contents)
+        self._send_output_to_pager(converted_content)
+
+    def _send_output_to_pager(self, output):
+        cmdline = self.get_pager_cmdline()
+        L.debug("Running command: %s", cmdline)
+        p = self._popen(cmdline, stdin=PIPE)
+        p.communicate(input=output)
+
+    def _popen(self, *args, **kwargs):
+        return subprocess.Popen(*args, **kwargs)
+
+    def _convert_doc_content(self, ctx, contents):
```

Comment:
Maybe add a bit of documentation to the methods that are here so that subclasses can override them